components/motor/gpio/loop.go

```diff
@@ -2,18 +2,102 @@ package gpio
 
 import (
 	"context"
+	"errors"
 
 	"go.viam.com/rdk/control"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+// TODO: RSDK-5610 test the scaling factor with a non-pi board with hardware pwm.
+var (
+	powerScaling     = 255.0
+	errConstantBlock = errors.New("constant block should be called 'set_point")
+	errEndpointBlock = errors.New("endpoint block should be called 'endpoint")
+	errTrapzBlock    = errors.New("trapezoidalVelocityProfile block should be called 'trapz")
+	errPIDBlock      = errors.New("PID block should be called 'PID")
 )
 
 // SetState sets the state of the motor for the built-in control loop.
 func (m *EncodedMotor) SetState(ctx context.Context, state []*control.Signal) error {
 	power := state[0].GetSignalValueAt(0)
-	return m.SetPower(ctx, power, nil)
+	// scale power input to the 0 to +/- 255 range from the control config
```

Comment:
nit: scale power to +/- 1 from PID output

Reply:
removing comment and adding a gain block